src/artpop/space/sample.py

```diff
@@ -20,35 +20,43 @@ def xy_from_grid(num_stars, model, xy_dim, sample_dim=None,
     ----------
     num_stars : int
         Number of stars (i.e., positions) to sample.
-    model : `~astropy.modeling.Fittable2DModel` 
-        A two-dimensional ``astropy`` model or a callable object that takes 
+    model : `~astropy.modeling.Fittable2DModel`
+        A two-dimensional ``astropy`` model or a callable object that takes
         x and y grid positions arguments: `model(xx, yy)`.
     xy_dim : int or list-like
         Dimensions of the mock image in xy coordinates. If int is given,
         will make the x and y dimensions the same.
     sample_dim : int or None, optional
-        Square dimension to sample within. This is useful if the mock image 
-        size is small compared to the scale of the model. To be useful, 
+        Square dimension to sample within. This is useful if the mock image
+        size is small compared to the scale of the model. To be useful,
         `sample_dim` should be larger than `xy_dim`.
     random_state : `None`, int, list of ints, or `~numpy.random.RandomState`
         If `None`, return the `~numpy.random.RandomState` singleton used by
         ``numpy.random``. If `int`, return a new `~numpy.random.RandomState`
         instance seeded with the `int`.  If `~numpy.random.RandomState`,
         return it. Otherwise raise ``ValueError``.
-    
+
     Returns
     -------
     xy : `~numpy.ma.MaskedArray`
-        Masked numpy array of xy positions. Positions that fall outside the 
+        Masked numpy array of xy positions. Positions that fall outside the
         mock image are masked.
     """
     xy_dim = check_xy_dim(xy_dim)
 
+    num_pix = 2 * sample_dim**2
+    byte_per_pixel = 8
+    memory_in_GB = (byte_per_pixel * num_pix) / 1e9
+    if memory_in_GB > 1:
+        msg = "Warning: This image will exceed 1 Gb in memory usage."
```

Comment:
Please remove  "Warning: ", since the logger already prints this when you use `logger.warning`. Also, change "This image" to "The sampling grid". Thanks!